tests/unittests/sources/test_azure.py

```diff
@@ -3378,6 +3378,7 @@ def test_poll_imds_returns_ovf_env(
                     method="GET",
                     timeout=dsaz.IMDS_TIMEOUT_IN_SECONDS,
                     url=full_url,
+                    stream=False,
```

Comment:
We are mocking `requests.Session.request` [1], in order to inspect the call args list of [2]. Therefore, it does make sense to receive `stream=False` there.

I agree it is counter-intuitive, because the tests mock out `resquest.Session` (which is two levels down under `url_helper.readurl`) instead of `url_helper.readurl`. But, I did not want to change the tests' philosophy in this PR.

[1] https://github.com/canonical/cloud-init/pull/1606/files/f8dc71638cf153ee15fcc5d8bee4779745e159ce#diff-6d641d364bf5bfef74ee6b8a94ac1a0ed5da320f3e5e203d3c9774254bedd7a9R3339
[2] https://github.com/canonical/cloud-init/pull/1606/files#diff-0518e1788fd623ef0c6f3372484f52dd8c28cc5390319e1eec62e28de061720bR319